crates/task-impls/src/transactions.rs

```diff
@@ -220,53 +217,25 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                             return None;
                         }
                     };
+
+                // encode the transactions
                 let encoded_transactions = match payload.encode() {
                     Ok(encoded) => encoded.into_iter().collect::<Vec<u8>>(),
                     Err(e) => {
                         error!("Failed to encode the block payload: {:?}.", e);
                         return None;
                     }
                 };
-                // TODO <https://github.com/EspressoSystems/HotShot/issues/1686>
-                let srs = test_srs(NUM_STORAGE_NODES);
-                // TODO We are using constant numbers for now, but they will change as the quorum size
-                // changes.
-                // TODO <https://github.com/EspressoSystems/HotShot/issues/1693>
-                let vid = VidScheme::new(NUM_CHUNKS, NUM_STORAGE_NODES, &srs).unwrap();
-                let vid_disperse = vid.disperse(encoded_transactions.clone()).unwrap();
-
-                // TODO never clone a block
-                // https://github.com/EspressoSystems/HotShot/issues/1858
+
+                // send the sequenced transactions to VID and DA tasks
                 self.event_stream
-                    .publish(HotShotEvent::BlockReady(
+                    .publish(HotShotEvent::TransactionsSequenced(
                         encoded_transactions,
-                        metadata,
+                        metadata.clone(),
```

Comment:
I don't see why we need a clone here. Can't the `TransactionsSequenced` event just take ownership of `metadata`?